src/image_checker.py

```diff
@@ -0,0 +1,285 @@
+            return (
+                False,
+                f"{image_name}: {self.redmine_error_color}tags list can not be obtain%",
+            )
+
+        potential_updates = self.filter_potential_image_updates(image_name, versions)
+        if not potential_updates:
+            return (
+                False,
+                f"{image_name}: {self.redmine_error_color}no similar flavour tag obtained%",
+            )
+
+        return True, potential_updates
+
+    def compare_versions(self, image, potential_updates, curr_version):
+        last_version = natsorted(potential_updates)[-1]
+
+        if last_version == curr_version:
+            return True, f"{image}:{curr_version}: {self.redmine_ok_color}Up to date%"
+        else:
+            curr_major = curr_version.split(".")[0]
+            last_major = last_version.split(".")[0]
+
+            if curr_major == last_major:
+                return (
+                    False,
+                    f"{image}:{curr_version}: {self.redmine_minor_color}minor upgrade to {last_version}%",
+                )
+            else:
+                status = f"{image}:{curr_version}: {self.redmine_major_color}major upgrade to {last_version}%"
+
+                same_major_updates = [
+                    v for v in potential_updates if v.split(".")[0] == curr_major
+                ]
+                if same_major_updates:
+                    last_minor_version = natsorted(same_major_updates)[-1]
+
+                    if last_minor_version and last_minor_version != curr_version:
+                        status += f"; {self.redmine_minor_color}minor upgrade to {image}:{last_minor_version}%"
+                return False, status
+
+    """
+        Returns a boolean (true = up to date, false = otherwise) and a message.
+    """
+
+    def check_image_status(self, image_name):
+        if image_name.startswith("docker.io/"):
+            image_name = image_name.replace("docker.io/", "")
+
+        if image_name in self.images_cache:
+            status, msg = self.images_cache[image_name]
+        else:
+            if ":" not in image_name or image_name.split(":")[1] == "latest":
+                status = False
+                msg = f"{image_name}: {self.redmine_error_color}can not check for updates to 'latest' tag%"
+            else:
+                success, versions = self.get_image_potential_updates(image_name)
+                if not success:
+                    status = False
+                    msg = versions
+                else:
+                    image, curr_version = image_name.split(":")
+                    status, msg = self.compare_versions(image, versions, curr_version)
+            self.images_cache[image_name] = (status, msg)
+
+        return status, msg
+
+    def check_base_image(self, image):
+        if not self.dockerhub_token:
+            return
+
+        if "/" not in image or not image.startswith("eeacms/"):
```

Comment:
I think the logic here is confusing, why not simply `if not image.startswith("eeacms/")`?